.github/workflows/kserve_m2m_test.yaml

```diff
@@ -162,58 +155,240 @@ jobs:
+
+        HTTP_CODE=${RESPONSE: -3}
+        RESPONSE_BODY=${RESPONSE:0:${#RESPONSE}-3}
+
+        if [[ "$HTTP_CODE" == "200" && "$RESPONSE_BODY" == *"predictions"* ]]; then
+          echo "Test passed: Previously unauthorized token now works with code $HTTP_CODE"
+        else
+          echo "Test failed: Previously unauthorized token still doesn't work with code $HTTP_CODE"
+          echo "Response: $RESPONSE_BODY"
+          exit 1
+        fi
+
+        # Restore original policy
+        kubectl apply -f - <<EOF
+        apiVersion: security.istio.io/v1
+        kind: AuthorizationPolicy
+        metadata:
+          name: cluster-local-gateway
+          namespace: istio-system
+        spec:
+          selector:
+            matchLabels:
+              app: cluster-local-gateway
+              istio: cluster-local-gateway
+          action: ALLOW
+          rules:
+          - to:
+            - operation:
+                hosts: ["*.kubeflow-user-example-com.svc.cluster.local"]
+            when:
+            - key: request.headers[kubeflow-userid]
+              values: ["system:serviceaccount:kubeflow-user-example-com:*"]
+          - to:
+            - operation:
+                hosts: ["*.kubeflow-user-example-com.svc.cluster.local"]
+            when:
+            - key: request.auth.claims[sub]
+              values: ["system:serviceaccount:kubeflow-user-example-com:*"]
+          - to:
+            - operation:
+                paths: ["/healthz", "/metrics", "/wait-for-drain"]
+        EOF
+
+    # Test 5: Internal access with token from same namespace
+    - name: Test internal access with token from same namespace
+      run: |
+        # Create a pod in the same namespace to test internal access
+        kubectl run -n kubeflow-user-example-com curl-test --image=curlimages/curl --restart=Never -- sleep 3600
+
+        # Wait for the pod to be ready
+        kubectl wait --for=condition=ready pod/curl-test -n kubeflow-user-example-com --timeout=60s
+
+        # Run test from inside the pod with token
+        RESULT=$(kubectl exec -n kubeflow-user-example-com curl-test -- \
+          curl -s -w "\n%{http_code}" \
+          -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+          -H "Content-Type: application/json" \
+          "http://isvc-sklearn-predictor.kubeflow-user-example-com.svc.cluster.local/v1/models/isvc-sklearn:predict" \
+          -d '{"instances": [[6.8, 2.8, 4.8, 1.4]]}')
+
+        HTTP_CODE=$(echo "$RESULT" | tail -n1)
+        RESPONSE_BODY=$(echo "$RESULT" | head -n1)
+
+        if [[ "$HTTP_CODE" == "200" && "$RESPONSE_BODY" == *"predictions"* ]]; then
+          echo "Test passed: Internal access with token successful with code $HTTP_CODE"
+        else
+          echo "Test failed: Internal access with token returned $HTTP_CODE"
+          echo "Response: $RESPONSE_BODY"
+          exit 1
+        fi
+
+    # Test 6: Internal access without token
```

Comment:
Why is there a comment that is just the name of the tests. Please remove such comments in general.